### PR TITLE
test: switch to `vitest` and reach 96% testing coverage

### PR DESCRIPTION
**Pull request by dan-online**
### Changes

- Switch to `vitest` from `jest`
- Fix push
- Fix autonum
- Fix ensure/autoensure

### CODE REVIEW

File: docs/api.md
Comment (eslachance) [inline docs/api.md:1]: Reset this file and ignore it from linters - it's autogenerated.

File: test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,956 @@
...
+    });
+
+    test('should create/use data dir', async () => {
+      await rm('./data', { recursive: true }).catch(() => {});
+
+      const enmap = new Enmap({ name: 'test' });
+      const enmap2 = new Enmap({ name: 'test' });
+
+      expect(enmap).toBeInstanceOf(Enmap);
+      expect(enmap2).toBeInstanceOf(Enmap);
```
Comment (eslachance) [inline test/index.spec.js:87]: why do we need 2 of those tests for the creation of the data dir? also, should we not actually check the datar dir exists here?

**Review by eslachance — CHANGES_REQUESTED** [review]

File: test/index.spec.js (hunk 1 above)
Comment (dan-online) [reply test/index.spec.js:87]: The second construct will hit the existsSync case, so it's essentially doing that for us

Comment (eslachance) [reply test/index.spec.js:87]: ah nice 👍🏻

**Review by eslachance — APPROVED** [review]

**Pull request merged**